### PR TITLE
Remove the pilot open feature flag

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

Removing the `pilot_open` feature flag as its been active in production for a while now and we have no intention of deactivating apply.

## Changes proposed in this pull request

- Remove all logic associated with the `pilot_open` feature flag 
- Remove the pilot holding page
- Rework some of the feature flag tests which previously used the `pilot_open` flag for the tests to use the `dfe_sign_in_fallback` feature flag
- Remove the `pilot_open` feature flag.
- Data migration to drop the `pilot_open` feature flag record.

## Guidance to review

Did I miss anything? 

## Link to Trello card

https://trello.com/c/NNEaghMZ/4486-remove-the-pilot-open-feature-flag

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: spec/components/previews/support_interface/feature_audit_trail_component_preview.rb [hunk 1]
```diff
@@ -1,9 +0,0 @@
-module SupportInterface
-  class FeatureAuditTrailComponentPreview < ViewComponent::Preview
-    def pilot_open
-      render SupportInterface::FeatureAuditTrailComponent.new(
-        feature: Feature.find_by(name: 'pilot_open'),
-      )
-    end
-  end
-end
```
Comment (stevehook) [inline spec/components/previews/support_interface/feature_audit_trail_component_preview.rb:9]: I think this file should stay but with a different example feature flag.

**Review by stevehook — COMMENTED** [review]
I think there may be a case for deploying the migration in a separate PR (or make it manual and run it a few minutes after the main deploy). This is because any request that's still running no the old code when the data migration has run could reinstate the `pilot_open` record. But we can leave as is I think because even if the flag is reinstated the invariant check will pick it up (or it can be checked manually shortly after deploy).

One other comment inline but looks great - lots of noise removed from the codebase!

File: spec/components/previews/support_interface/feature_audit_trail_component_preview.rb (hunk 1 above)
Comment (avinhurry) [reply spec/components/previews/support_interface/feature_audit_trail_component_preview.rb]: Good spot! Done ♻️

**Review by stevehook — APPROVED** [review]
Good stuff. Let's ship this one. We can double check that the feature flag record stayed deleted after the deploy.

**Pull request merged**